src/OpenTelemetry.Instrumentation.Dependencies/OpenTelemetryBuilderExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="OpenTelemetryBuilderExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Instrumentation.Dependencies.Implementation;
+
+namespace OpenTelemetry.Trace.Configuration
+{
+    /// <summary>
+    /// Extension methods to simplify registering of dependency instrumentation.
+    /// </summary>
+    public static class OpenTelemetryBuilderExtensions
+    {
+        /// <summary>
+        /// Enables the outgoing requests automatic data collection for all supported activity sources.
+        /// </summary>
+        /// <param name="builder"><see cref="OpenTelemetryBuilder"/> being configured.</param>
+        /// <returns>The instance of <see cref="OpenTelemetryBuilder"/> to chain the calls.</returns>
+        public static OpenTelemetryBuilder AddDependencyInstrumentation(this OpenTelemetryBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+#if NET461
+            builder.AddHttpWebRequestDependencyInstrumentation();
+#endif
+            return builder;
+        }
+
+#if NET461
+        /// <summary>
+        /// Enables the outgoing requests automatic data collection for .NET Framework HttpWebRequest activity source.
+        /// </summary>
+        /// <param name="builder"><see cref="OpenTelemetryBuilder"/> being configured.</param>
+        /// <returns>The instance of <see cref="OpenTelemetryBuilder"/> to chain the calls.</returns>
+        public static OpenTelemetryBuilder AddHttpWebRequestDependencyInstrumentation(this OpenTelemetryBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            GC.KeepAlive(HttpWebRequestActivitySource.Instance);
```

Comment:
Q.: Why is this needed?

Reply:
To be honest, not totally sure. That came from the original version in dotnet/runtime:

https://github.com/dotnet/runtime/blob/bd4697644e7024d1e5be94b674f73066bee112a4/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs#L40

I think it is basically a clever way of telling the runtime to initialize the static stuff in the HttpWebRequestActivitySource class. We need to touch some code for the runtime to do that, but we don't actually want to call anything. Doesn't really have anything to do with garbage collection AFAIK.

Reply:
Thanks. It feels confusing because it references `GC`, perhaps it should be a method instead: `Init()` or `Initialize()` - anyway I think it is fine to address it later.

Reply:
The Gc.Keepalive was part of the existing code, not introduced in this PR. Adding this note here and proceeding to merge this PR.